### PR TITLE
mk DR-880 share snapshot during creation 

### PR DESCRIPTION
**Pull request by mackenziekerwin**
share-panel
One major change:
This share panel slides out in the query builder to add readers _before_ creating the snapshot (basically edits `snapshot.readers` in the store).
And one little change:
The `ADD_SNAPSHOT_POLICY_MEMBER` action that uses the api endpoint now supports adding discoverers as well. I realized after making the change that this is different from sharing before snapshot creation, but I kept those changes in because this is probably getting implemented soon.

### CODE REVIEW

File: cypress/integration/queryBuilder.spec.js [hunk 1]
```diff
@@ -30,4 +30,18 @@ describe('test query builder', () => {
     cy.get('[data-cy="applyFiltersButton"]').click();
     cy.get('[data-cy=appliedFilterList-ancestry_specific_meta_analysis]', {}).should('be.visible');
   });
+
+  it.only('adds/removes readers', () => {
```
Comment (myessail) [inline cypress/integration/queryBuilder.spec.js]: make sure you remove the `.only` before you merge

File: package-lock.json [hunk 2]
```diff
@@ -3107,7 +3107,7 @@
         "source-map": {
           "version": "0.6.1",
           "resolved": "https://registry.npmjs.org/source-map/-/source-map-0.6.1.tgz",
-          "integrity": "sha512-UjgapumWlbMhkBgzT7Ykc5YXUT46F0iKu8SGXq0bcwP5dz/h0Plj6enJqjz1Zbq2l5WaqYnrVbwWOWMyF3F47g==",
+          "integrity": "sha1-dHIq8y6WFOnCh6jQu95IteLxomM=",
```
Comment (myessail) [inline package-lock.json]: i would remove all changes to this file (`git checkout develop package-lock.json`), it looks like we didn't make any package changes here so no need to change the lockfile

File: src/components/dataset/query/sidebar/panels/ShareSnapshot.jsx [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+  /**
+   * adds the email as a tag to the text box
+   */
+  inputEmail = (event, value) => {
+    this.setState({ usersToAdd: value, currentInput: '' });
+  };
+
+  /**
+   * updates the snapshot readers to include users that have been entered
```
Comment (myessail) [inline src/components/dataset/query/sidebar/panels/ShareSnapshot.jsx:107]: yay comments!!

**Review by myessail — APPROVED** [review]
Looks great! Few comments before you merge:

**Review by thathert — COMMENTED** [review]
LGTM!

**Review by fboulnois — APPROVED** [review]
lgtm 😃

File: cypress/integration/queryBuilder.spec.js [hunk 4]
```diff
@@ -30,4 +30,18 @@ describe('test query builder', () => {
...
   });
+
+  it('adds/removes readers', () => {
+    // selects the share button in the sidebar
+    cy.get('div.MuiButtonBase-root:nth-child(3) > svg:nth-child(1)').click();
+    cy.get('[data-cy=enterEmailBox]').type('mkerwin.dev@gmail.com');
+    cy.get('[data-cy=inviteButton]').click();
+    cy.get('[data-cy=readers]')
+      .contains('mkerwin.dev@gmail.com')
+      .should('be.visible');
```
Comment (thathert) [inline cypress/integration/queryBuilder.spec.js:41]: How about a test to type in a comma between emails and see if you get two entries?

**Review by thathert — APPROVED** [review]
One request

File: cypress/integration/queryBuilder.spec.js (hunk 4 above)
Comment (mackenziekerwin) [reply cypress/integration/queryBuilder.spec.js:41]: done!

**Pull request merged**